experimental/dds/tree2/src/shared-tree/sharedTreeView.ts

```diff
@@ -55,6 +56,11 @@ export interface ViewEvents {
 	 * This is mainly useful for knowing when to do followup work scheduled during events from Anchors.
 	 */
 	afterBatch(): void;
+
+	/**
+	 * An undoable change has been made to the tree. This is used to track undo/redo on the {@link SharedTreeViewUndoRedoHandler}
```

Comment:
Are you sure you want to publically export this event (and also have to export `UndoRedoManagerCommitType`), solely to support an unmaintained legacy framework feature? Would you want/expect a user to actually subscribe to this event, or does it exist just so that the UndoRedoHandler can listen to it?

You could:

1. Hide it publically but not internally by using another interface (`ViewEventsInternal`, etc.).

or

2. Make it less specific, e.g. change it to a "change" event, which also passes a commit type (in which case, consider renaming `UndoRedoManagerCommitType` to something like `RevertType` and moving it higher up into utils).

or

3. Remove this event and instead expose a method on `ISharedTreeView` called `createUndoRedoHandler(manager: UndoRedoStackManager)`. It hooks itself up to the branch `change/undoable` event as it's created by the view. SharedTree already has an inverted undo/redo dependency (it depends on the fluid undo/redo package whereas for a DDS like map I believe it's the other way around) so it doesn't seem too inappropriate for SharedTree's `UndoRedoHandlers` to be create in a different way than other DDS_s.

Reply:
Yeah, our conversation earlier convinced me we don't want to even expose the revertible so I think hiding/doing something else with this event as well makes sense.